library/std/src/fs.rs

```diff
@@ -116,6 +116,13 @@ pub struct File {
     inner: fs_imp::File,
 }
 
+#[unstable(feature = "dirfd", issue = "120426")]
+#[cfg(target_family = "unix")]
```

Comment:
We shouldn't/can't have platform-gated config in `fs`, tidy would probably complain about this if CI got that far. It's okay to stub out less popular platforms for now with `FIXME`s, but I think the initial implementation should come with Windows support. There are some guidelines at https://github.com/rust-lang/rust/issues/120426#issuecomment-1913167876, feel free to ask questions on Zulip if you get stuck.